pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -383,6 +376,19 @@ ssize_t sock_dtls_send(sock_dtls_t *sock, sock_dtls_session_t *remote,
                       len);
 }
 
+static ssize_t _get_buffer(sock_dtls_t *sock, void *data, size_t max_len)
+{
+    uint8_t *buf = sock->buf;
+    size_t buflen = sock->buflen;
+
+    sock->buf = NULL;
+    if (buflen > max_len) {
+        return -ENOBUFS;
+    }
+    memcpy(data, buf, buflen);
```

Comment:
this was a memmove before,  is it safe to change that? Kind of unrelated to this PR as it was like that before, but this also made me think: is it at all safe to use the same memory location to first hold the received udp data, then hand it to `dtls_handle_message` *and then* use the same memory as target buffer for the encrypted data. I.e. not thinking "crypto-safe" but more "memory safe": i.e. is tinydtls actually able to use the same buffer as source and target buffer for the decrypt?

Reply:
I use `memmove()` because there have been problem due to the buffer overlapping.

> [...] is tinydtls actually able to use the same buffer as source and target buffer for the decrypt?

I took a look at the tinydtls code and it uses the same buffer as source and target buffer for the decrypt, just with offset for the decrypted data to skip the headers.

Reply:
So, keep `memmove()`?

Reply:
Done in 1c772a0 with a comment.